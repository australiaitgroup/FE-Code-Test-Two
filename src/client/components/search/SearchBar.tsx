import * as React from 'react';
import { Button, Input } from 'semantic-ui-react';

import { ISearchBarProps } from '../interfaces/searchBar.interface';

let searchTimer;

const SearchBar: React.FunctionComponent<ISearchBarProps> = ({
    handleInputChange,
    search,
}) => (
    <Input
        action={
            <Button
                color="teal"
                content="Search"
                onClick={search}
            />
        }
        fluid
        icon="search"
        iconPosition="left"
        onKeyUp={(event) => {
            clearTimeout(searchTimer);
            handleInputChange(event.target.value);
            if (event.key === 'Enter') return search();
            searchTimer = setTimeout(search, 300);
        }}
        placeholder="Search..."
    />
);

export default SearchBar;
