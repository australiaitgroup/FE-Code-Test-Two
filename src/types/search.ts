export interface ISearchResult {
    hits: IHit[];
    nbHits: number;
    page: number;
    nbPages: number;
    hitsPerPage: number;
    processingTimeMS: number;
    exhaustiveNbHits: boolean;
    query: string;
    params: string;
}

export interface IHit {
    created_at: string;
    title: string | null;
    url: string | null;
    author: string;
    points: number;
    story_text: string | null;
    comment_text: string | null;
    num_comments: number | null;
    story_id: number | null;
    story_title: string | null;
    story_url: string | null;
    parent_id: number | null;
    created_at_i: number;
    _tags: string[];
    objectID: string;
    _highlightResult: {
        author: IHightlightResultItem;
        comment_text: IHightlightResultItem;
        story_text: IHightlightResultItem;
        story_title: IHightlightResultItem,
        title: IHightlightResultItem;
        url: IHightlightResultItem;
    };
}

interface IHightlightResultItem {
    value: string;
    matchLevel: string;
    fullyHighlighted?: boolean;
    matchedWords: string[];
}

export const enum SearchType {
    All = 'All',
    Stories = 'Stories',
    Comments = 'Comments',
}

export const enum SortOrder {
    Popularity = 'Popularity',
    Date = 'Date',
}

export const enum DateRange {
    AllTime = 'All Time',
    LastDay = 'Last 24 hours',
    PastWeek = 'Past Week',
    PastMonth = 'Past Month',
    PastYear = 'Past Year',
    Custom = 'Custom',
}
