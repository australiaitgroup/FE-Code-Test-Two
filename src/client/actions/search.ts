import fetch from '../utils/fetch';
import generateURL from '../utils/searchUrlHelpers';
import { ISearchResult } from '../../types/search';

// action types

export const SEARCH = 'SEARCH';
export const HANDLE_SEARCH_RESULT = 'HANDLE_SEARCH_RESULT';
export const HANDLE_INPUT_CHANGE = 'HANDLE_INPUT_CHANGE';
export const HANDLE_PAGE_CHANGE = 'HANDLE_PAGE_CHANGE';
export const SET_SEARCH_ERROR = 'SET_SEARCH_ERROR';

// action creators
â€‹
export function searchAction(searchFor: string) {
    return { searchFor, type: SEARCH };
}

export function handleSearchResultAction(result: ISearchResult) {
    return { result, type: HANDLE_SEARCH_RESULT };
}

export function handleInputChangeAction(input: string) {
    return { input, type: HANDLE_INPUT_CHANGE };
}

export function handlePageChangeAction(page: number) {
    return { page, type: HANDLE_PAGE_CHANGE };
}

export function setSearchErrorAction(error: Error) {
    return { error, type: SET_SEARCH_ERROR };
}

// async actions

export function doSearch(page?: number) {
    return (dispatch, getState) => {
        const input = getState().search.input;
        const settings = getState().settings;
        const url = generateURL(settings, input, page);

        if (page) {
            dispatch(handlePageChangeAction(page));
        } else {
            dispatch(searchAction(input));
        }

        return fetch(url)
            .then(result => dispatch(handleSearchResultAction(result)))
            .catch(err => dispatch(setSearchErrorAction(err)));
    };
}
