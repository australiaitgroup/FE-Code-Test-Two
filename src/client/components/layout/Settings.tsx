import 'react-datepicker/dist/react-datepicker.css';

import * as moment from 'moment';
import * as React from 'react';
import DatePicker from 'react-datepicker';
import { Button, Menu, Select } from 'semantic-ui-react';

import { ISettingsProps } from '../interfaces/settings.interface';
import { DateRange, SearchType, SortOrder } from '../../../types/search';
import * as s from './Layout.css';

const Settings: React.FunctionComponent<ISettingsProps> = ({
    apply,
    dateRange,
    from,
    hitsPerPage,
    searchType,
    setState,
    sortOrder,
    to,
    toggle,
}) => (
    <React.Fragment>
        <Menu.Item>
            <p>Search type</p>
            <Select
                onChange={handleChange(setState, 'searchType')}
                options={[
                    { text: 'All', value: SearchType.All },
                    { text: 'Stories', value: SearchType.Stories },
                    { text: 'Comments', value: SearchType.Comments },
                ]}
                value={searchType}
            />
        </Menu.Item>
        <Menu.Item>
            <p>Sort order</p>
            <Select
                onChange={handleChange(setState, 'sortOrder')}
                options={[
                    { text: 'Popularity', value: SortOrder.Popularity },
                    { text: 'Date', value: SortOrder.Date },
                ]}
                value={sortOrder}
            />
        </Menu.Item>
        <Menu.Item fitted="horizontally">
            <p>Date range</p>
            <Select
                onChange={handleChange(setState, 'dateRange')}
                options={[
                    { text: 'All time', value: DateRange.AllTime },
                    { text: 'Last 24h', value: DateRange.LastDay },
                    { text: 'Past Week', value: DateRange.PastWeek },
                    { text: 'Past Month', value: DateRange.PastMonth },
                    { text: 'Past Year', value: DateRange.PastYear },
                    { text: 'Custom range', value: DateRange.Custom },
                ]}
                value={dateRange}
            />
            {
                dateRange === DateRange.Custom
                    ? (
                        <React.Fragment>
                            <p>From</p>
                            <DatePicker
                                className={s.minWidth}
                                dateFormat="YYYY-MM-DD"
                                onChange={date => setState({ from: date.format('YYYY-MM-DD') })}
                                selected={moment(from)}
                            />
                            <p>To</p>
                            <DatePicker
                                className={s.minWidth}
                                dateFormat="YYYY-MM-DD"
                                onChange={date => setState({ to: date.format('YYYY-MM-DD') })}
                                selected={moment(to)}
                            />
                        </React.Fragment>
                    )
                    : null
            }
        </Menu.Item>
        <Menu.Item>
            <p>Hits per page</p>
            <Select
                onChange={handleChange(setState, 'hitsPerPage')}
                options={[
                    { text: '10', value: 10 },
                    { text: '20', value: 20 },
                    { text: '30', value: 30 },
                    { text: '50', value: 50 },
                ]}
                value={hitsPerPage}
            />
        </Menu.Item>
        <Menu.Item>
            <Button
                onClick={apply}
                color="green"
                size="mini"
            >
                Apply
            </Button>
            <Button
                onClick={toggle}
                size="mini"
            >
                Cancel
            </Button>
        </Menu.Item>
    </React.Fragment>
);

function handleChange(setState: ISettingsProps['setState'], key: string) {
    return (event, data) => {
        setState({ [key]: data.value });
    };
}

export default Settings;
