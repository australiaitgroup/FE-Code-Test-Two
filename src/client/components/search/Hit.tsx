import * as moment from 'moment';
import * as React from 'react';
import MediaQuery from 'react-responsive';
import { Icon, Item, Label } from 'semantic-ui-react';

import { IHit } from '../../../types/search';
import * as s from './hit/Hit.css';

const Hit: React.FunctionComponent<{ data: IHit }> = ({ data }) => (
    <Item>
        <Item.Content>
            <Item.Header
                as={data.url || data.story_url ? 'a' : null}
                href={data.url || data.story_url}
            >
                <p
                    dangerouslySetInnerHTML={{ // Assume the response is sanitised
                        __html: (data._highlightResult.title
                                && data._highlightResult.title.value)
                            || (data._highlightResult.story_title
                                && data._highlightResult.story_title.value)
                            || 'Title not found',
                    }}
                />
            </Item.Header>
            <MediaQuery maxWidth={768}>
                {(matches) => {
                    const size = matches ? 'mini' : 'medium';
                    return (
                        <Item.Meta>
                            <Label className={s.marginTop} size={size}>
                                {data.points || 0} points
                            </Label>
                            <Label className={s.marginTop} size={size}>
                                <Icon name="user" />
                                <span
                                    dangerouslySetInnerHTML={{
                                        __html: data._highlightResult.author
                                            && data._highlightResult.author.value,
                                    }}
                                />
                            </Label>
                            <Label className={s.marginTop} size={size}>
                                {moment(data.created_at).fromNow()}
                            </Label>
                            <Label className={s.marginTop} size={size}>
                                {data.num_comments ? `${data.num_comments} comments` : 'comment'}
                            </Label>
                            {
                                data.url && !matches
                                    ? (
                                        <Label
                                            as="a"
                                            className={s.marginTop}
                                            href={data.url}
                                            size={size}
                                        >
                                            <span
                                                dangerouslySetInnerHTML={{
                                                    __html: data._highlightResult.url
                                                        && data._highlightResult.url.value,
                                                }}
                                            />
                                        </Label>
                                    )
                                    : null
                            }
                        </Item.Meta>
                    );
                }}
            </MediaQuery>
            <Item.Description>
                {
                    data._highlightResult.comment_text || data._highlightResult.story_text
                        ? (
                            <div
                                dangerouslySetInnerHTML={{
                                    __html: (data._highlightResult.comment_text
                                            && data._highlightResult.comment_text.value)
                                        || (data._highlightResult.story_text
                                            && data._highlightResult.story_text.value),
                                }}
                            />
                        )
                        : null
                }
            </Item.Description>
        </Item.Content>
    </Item>
);

export default Hit;
