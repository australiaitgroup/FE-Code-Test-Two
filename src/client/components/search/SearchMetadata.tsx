import * as React from 'react';
import MediaQuery from 'react-responsive';
import { List, Message } from 'semantic-ui-react';

import { ISearchMetadataProps } from '../interfaces/searchMetadata.interface';
import { DateRange } from '../../../types/search';

const SearchMetadata: React.FunctionComponent<ISearchMetadataProps> = ({ nbHits, settings }) => (
    nbHits !== undefined
        ? (
            <MediaQuery minWidth={768}>
                {(matches) => {
                    const [floated, horizontal, size] = matches
                        ? ['right', true, 'small']
                        : [undefined, undefined, 'tiny'] as any;

                    return (
                        <Message info size={size}>
                            <span>{nbHits === 0 ? 'No' : nbHits} results found</span>
                            <List
                                floated={floated}
                                horizontal={horizontal}
                            >
                                <List.Item>Search Type: {settings.searchType}</List.Item>
                                <List.Item>Sort Order: {settings.sortOrder}</List.Item>
                                <List.Item>
                                    Date Range: {
                                    settings.dateRange !== DateRange.Custom
                                        ? settings.dateRange
                                        : `${settings.from} To ${settings.to}`
                                }
                                </List.Item>
                                <List.Item>Hits Per Page: {settings.hitsPerPage}</List.Item>
                            </List>
                        </Message>
                    );
                }}
            </MediaQuery>
        )
        : null
);

export default SearchMetadata;
