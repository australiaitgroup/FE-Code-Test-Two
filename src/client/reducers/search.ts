import {
    HANDLE_INPUT_CHANGE,
    HANDLE_PAGE_CHANGE,
    HANDLE_SEARCH_RESULT,
    SEARCH,
    SET_SEARCH_ERROR,
} from '../actions/search';
import { ISearchResult } from '../../types/search';
import { IAction, ISearchState } from '../../types/stateAndAction';

const initialState = {
    error: null,
    isFetching: false,
    input: '',
    page: 1,
    result: {} as ISearchResult,
    searchFor: '',
};

export default function reducer(
    state: ISearchState = initialState,
    action: IAction,
): ISearchState {
    switch (action.type) {
        case SEARCH:
            return {
                ...state,
                error: null,
                isFetching: true,
                page: 1,
                result: {} as ISearchResult,
                searchFor: action.searchFor,
            };

        case HANDLE_SEARCH_RESULT:
            if (state.searchFor !== action.result.query) return state;
            return { ...state, isFetching: false, result: action.result };

        case HANDLE_INPUT_CHANGE:
            return { ...state, input: action.input };

        case HANDLE_PAGE_CHANGE:
            return { ...state, isFetching: true, page: action.page };

        case SET_SEARCH_ERROR:
            return { ...state, error: action.error, isFetching: false };

        default:
            return state;
    }
}
