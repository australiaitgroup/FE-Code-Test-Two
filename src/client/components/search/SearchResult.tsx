import * as React from 'react';
import { Header, Icon, Item, Message, Segment } from 'semantic-ui-react';

import { ISearchResultProps } from '../interfaces/searchResult.interface';
import Placeholder from '../Placeholder';
import Hit from './Hit';

const SearchResult: React.FunctionComponent<ISearchResultProps> = ({
    error,
    hits,
    isFetching,
    nbHits,
}) => {
    const noMatch = !isFetching && nbHits === 0;

    const getContent = () => {
        if (isFetching) return <Placeholder />;

        if (error) {
            return (
                <Message negative>
                    <Message.Header>Sorry, there was an error</Message.Header>
                    <p>{error.message}</p>
                </Message>
            );
        }

        if (noMatch) {
            return (
                <Header icon>
                    <Icon name="search" />
                    No results matching your query.
                </Header>
            );
        }

        return (
            <Item.Group divided>
                {(hits || []).map(hit => <Hit key={hit.objectID} data={hit} />)}
            </Item.Group>
        );
    };

    return (
        <Segment loading={isFetching} placeholder={noMatch}>
            {getContent()}
        </Segment>
    );
};

export default SearchResult;
