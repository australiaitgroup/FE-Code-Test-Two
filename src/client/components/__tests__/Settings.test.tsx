import * as React from 'react';

import Settings from '../layout/Settings';
import { DateRange, SearchType, SortOrder } from '../../../types/search';
import { renderComponent } from '../../../../tools/testHelpers/componentHelpers';

const props = {
    apply: () => Promise.resolve(),
    dateRange: DateRange.PastWeek,
    from: '2018-11-20',
    hitsPerPage: 30,
    searchType: SearchType.Comments,
    setState: () => {},
    sortOrder: SortOrder.Popularity,
    to: '2018-11-21',
    toggle: () => {},
};

describe.each([
    [
        props,
        'normal date range',
    ],
    [
        {
            ...props,
            dateRange: DateRange.Custom,
        },
        'custom date range',
    ],
])(
    'Settings renders correctly',
    (props, name) => {
        it(`(${name})`, () => {
            const component = renderComponent(<Settings {...props} />, {});
            const result = component.toJSON();
            expect(result).toMatchSnapshot();
        });
    },
);
