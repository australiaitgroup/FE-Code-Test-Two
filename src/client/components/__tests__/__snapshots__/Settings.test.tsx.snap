// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders correctly (custom date range) 1`] = `
Array [
  <div
    className="item"
    onClick={[Function]}
  >
    <p>
      Search type
    </p>
    <div
      aria-expanded={false}
      className="ui selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Comments
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            All
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Stories
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Comments
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="item"
    onClick={[Function]}
  >
    <p>
      Sort order
    </p>
    <div
      aria-expanded={false}
      className="ui selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Popularity
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Popularity
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Date
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="horizontally fitted item"
    onClick={[Function]}
  >
    <p>
      Date range
    </p>
    <div
      aria-expanded={false}
      className="ui selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Custom range
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            All time
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Last 24h
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Past Week
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Past Month
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Past Year
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Custom range
          </span>
        </div>
      </div>
    </div>
    <p>
      From
    </p>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          className="minWidth"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value="2018-11-20"
        />
      </div>
    </div>
    <p>
      To
    </p>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          className="minWidth"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value="2018-11-21"
        />
      </div>
    </div>
  </div>,
  <div
    className="item"
    onClick={[Function]}
  >
    <p>
      Hits per page
    </p>
    <div
      aria-expanded={false}
      className="ui selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        30
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            10
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            20
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            30
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            50
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="item"
    onClick={[Function]}
  >
    <button
      className="ui green mini button"
      onClick={[Function]}
      role="button"
    >
      Apply
    </button>
    <button
      className="ui mini button"
      onClick={[Function]}
      role="button"
    >
      Cancel
    </button>
  </div>,
]
`;

exports[`Settings renders correctly (normal date range) 1`] = `
Array [
  <div
    className="item"
    onClick={[Function]}
  >
    <p>
      Search type
    </p>
    <div
      aria-expanded={false}
      className="ui selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Comments
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            All
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Stories
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Comments
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="item"
    onClick={[Function]}
  >
    <p>
      Sort order
    </p>
    <div
      aria-expanded={false}
      className="ui selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Popularity
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Popularity
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Date
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="horizontally fitted item"
    onClick={[Function]}
  >
    <p>
      Date range
    </p>
    <div
      aria-expanded={false}
      className="ui selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Past Week
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            All time
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Last 24h
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Past Week
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Past Month
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Past Year
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Custom range
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="item"
    onClick={[Function]}
  >
    <p>
      Hits per page
    </p>
    <div
      aria-expanded={false}
      className="ui selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        30
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            10
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            20
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            30
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            50
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="item"
    onClick={[Function]}
  >
    <button
      className="ui green mini button"
      onClick={[Function]}
      role="button"
    >
      Apply
    </button>
    <button
      className="ui mini button"
      onClick={[Function]}
      role="button"
    >
      Cancel
    </button>
  </div>,
]
`;
